src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/CoolerListener.java

```diff
@@ -47,14 +49,29 @@ public void onHungerLoss(FoodLevelChangeEvent e) {
         Player p = (Player) e.getEntity();
 
         if (e.getFoodLevel() < p.getFoodLevel()) {
-            for (ItemStack item : p.getInventory().getContents()) {
-                if (cooler.isItem(item)) {
-                    if (Slimefun.hasUnlocked(p, cooler, true)) {
-                        takeJuiceFromCooler(p, item);
-                    }
-                    else {
-                        return;
-                    }
+            checkAndConsume(p);
+        }
+    }
+    
+    @EventHandler
+    public void onHungerDamage(EntityDamageEvent e) {
+        if (cooler == null || cooler.isDisabled()) {
+            return;
+        }
+        
+        if (e.getCause() == EntityDamageEvent.DamageCause.STARVATION) {
+            checkAndConsume((Player) e.getEntity());
```

Comment:
You will need to check whether the Entity is actually a Player first before casting.